src/commands/Moderation/prune.ts

```diff
@@ -20,59 +51,102 @@ export default class extends SkyraCommand {
 			flagSupport: true,
 			requiredPermissions: ['MANAGE_MESSAGES', 'READ_MESSAGE_HISTORY'],
 			runIn: ['text'],
-			usage: '[limit:integer{1,100}] [link|invite|bots|you|me|upload|user:user]',
+			usage: '[limit:integer{1,100}] (position:position) (message:message) [filter:filter|user:user]',
 			usageDelim: ' '
 		});
+
+		this.createCustomResolver('position', (argument, _possible, message) => {
+			if (!argument) return null;
+			const position = this.kPositions.get(argument.toLowerCase());
+			if (typeof position === 'undefined') throw message.language.tget('COMMAND_PRUNE_INVALID_POSITION');
+			return position;
+		}).createCustomResolver('message', async (argument, possible, message, [, position]) => {
+			if (position === null) return message;
+
+			const fetched = this.kMessageRegExp.test(argument) ? await message.channel.messages.fetch(argument).catch(() => null) : null;
```

Comment:
You could also add support for message links, but that might require a regex change in klasa core

Reply:
No, it leaves margin for many errors, specially when it points to another channel.

Reply:
Aaand you can then check if the channel ID matches the current channel? 
```ts
const match = argument.match(/https?:\/\/(?:(ptb|canary)\.)?discordapp\.com\/channels\/\d+\/(\d+)\/(\d+)/);
if (match) {
	const [channelID, messageID] = match;
	if (channelID !== message.channel.id) throw 'bad';
}
```
Doesn't seem TOO trivial. This is, again, just a suggestion, not a must do 😄

Reply:
You can make a PR later adding this, but for now, I want to get this PR finished.